exporter/splunkhecexporter/metricdata_to_splunk.go

```diff
@@ -114,119 +151,43 @@ func metricDataToSplunk(logger *zap.Logger, data pdata.Metrics, config *Config)
-		// at indexing time, which will be much more useful than a 0-epoch-time value.
-		return nil
-	}
-
-	val := float64(seconds) + math.Round(float64(nanos)/1e6)/1e3
-
-	return &val
 }
 
-func mapValues(logger *zap.Logger, metric *metricspb.Metric, value interface{}) (map[string]interface{}, error) {
-	metricName := fmt.Sprintf("%s:%s", splunkMetricValue, metric.GetMetricDescriptor().Name)
-	switch pv := value.(type) {
-	case *metricspb.Point_Int64Value:
-		return map[string]interface{}{metricName: pv.Int64Value}, nil
-	case *metricspb.Point_DoubleValue:
-		return map[string]interface{}{metricName: pv.DoubleValue}, nil
-	case *metricspb.Point_DistributionValue:
-		return mapDistributionValue(metric, pv.DistributionValue)
-	case *metricspb.Point_SummaryValue:
-		return mapSummaryValue(metric, pv.SummaryValue)
-	default:
-		return nil, errors.New("unsupported metric type")
-	}
+func populateLabels(fields map[string]interface{}, labelsMap pdata.StringMap) {
+	labelsMap.ForEach(func(k string, v string) {
+		fields[k] = v
+	})
 }
 
-func mapDistributionValue(metric *metricspb.Metric, distributionValue *metricspb.DistributionValue) (map[string]interface{}, error) {
-	// Translating distribution values per symmetrical recommendations to Prometheus:
-	// https://docs.signalfx.com/en/latest/integrations/agent/monitors/prometheus-exporter.html#overview
-
-	// 1. The total count gets converted to a cumulative counter called
-	// <basename>_count.
-	// 2. The total sum gets converted to a cumulative counter called <basename>.
-	values := make(map[string]interface{})
-	metricName := fmt.Sprintf("%s:%s", splunkMetricValue, metric.GetMetricDescriptor().Name)
-	values[metricName+separator+countSuffix] = distributionValue.Count
-	values[metricName] = distributionValue.Sum
-	values[metricName+separator+sumOfSquaredDeviation] = distributionValue.SumOfSquaredDeviation
-
-	// 3. Each histogram bucket is converted to a cumulative counter called
-	// <basename>_bucket and will include a dimension called upper_bound that
-	// specifies the maximum value in that bucket. This metric specifies the
-	// number of events with a value that is less than or equal to the upper
-	// bound.
-	bucketMetricName := metricName + separator + bucketSuffix + separator
-	explicitBuckets := distributionValue.BucketOptions.GetExplicit()
-	if explicitBuckets == nil {
-		return values, fmt.Errorf(
-			"unknown bucket options type for metric %q",
-			bucketMetricName)
-	}
-	bounds := explicitBuckets.Bounds
-	splunkBounds := make([]*string, len(bounds)+1)
-	for i := 0; i < len(bounds); i++ {
-		dimValue := float64ToDimValue(bounds[i])
-		splunkBounds[i] = &dimValue
-	}
-	splunkBounds[len(splunkBounds)-1] = &infinityBoundSFxDimValue
-
-	for i, bucket := range distributionValue.Buckets {
-		values[fmt.Sprintf("%s%s", bucketMetricName, *splunkBounds[i])] = bucket.Count
+func copy(fields map[string]interface{}) map[string]interface{} {
```

Comment:
Nit: usually function named `copy` uses a source and destination. When the function _creates_ a copy usually it is named `clone`.

Reply:
ok changed to `cloneMap`